ledger-service/http-json/src/main/scala/com/digitalasset/http/Endpoints.scala

```diff
@@ -66,64 +67,102 @@ class Endpoints(
   import util.ErrorOps._
   import Uri.Path._
 
+  // Parenthesis in the case matches below are required because otherwise scalafmt breaks.
+  //noinspection ScalaUnnecessaryParentheses
   def all(implicit
       lc: LoggingContextOf[InstanceUUID],
       metrics: Metrics,
   ): PartialFunction[HttpRequest, Http.IncomingConnection => Future[HttpResponse]] = {
-    val dispatch: PartialFunction[HttpRequest, LoggingContextOf[
-      InstanceUUID with RequestID
-    ] => Future[HttpResponse]] = {
-      // Parenthesis are required because otherwise scalafmt breaks.
-      case req @ HttpRequest(POST, Uri.Path("/v1/create"), _, _, _) =>
-        (implicit lc => httpResponse(create(req)))
-      case req @ HttpRequest(POST, Uri.Path("/v1/exercise"), _, _, _) =>
-        (implicit lc => httpResponse(exercise(req)))
-      case req @ HttpRequest(POST, Uri.Path("/v1/create-and-exercise"), _, _, _) =>
-        (implicit lc => httpResponse(createAndExercise(req)))
-      case req @ HttpRequest(POST, Uri.Path("/v1/fetch"), _, _, _) =>
-        (implicit lc => httpResponse(fetch(req)))
+    val apiMetrics = metrics.daml.HttpJsonApi
+    type DispatchFun =
+      PartialFunction[HttpRequest, LoggingContextOf[InstanceUUID with RequestID] => Future[
+        HttpResponse
+      ]]
+    def mkDispatchFunWithTimer(timer: Timer)(fun: DispatchFun): DispatchFun =
+      fun andThen (f => lc => Timed.future(timer, f(lc)))
+    val commandDispatch =
+      mkDispatchFunWithTimer(apiMetrics.commandSubmissionTimer) {
+        case req @ HttpRequest(POST, Uri.Path("/v1/create"), _, _, _) =>
+          (implicit lc => httpResponse(create(req)))
+        case req @ HttpRequest(POST, Uri.Path("/v1/exercise"), _, _, _) =>
+          (implicit lc => httpResponse(exercise(req)))
+        case req @ HttpRequest(POST, Uri.Path("/v1/create-and-exercise"), _, _, _) =>
+          (implicit lc => httpResponse(createAndExercise(req)))
+      }
+    val queryDispatch = mkDispatchFunWithTimer(apiMetrics.queryTimer) {
```

Comment:
We probably should have separate timers for GET and POST here. The GET wildcard request which fetches all templates bypasses the database cache so we cannot really expect that it correlates with the POST requests which are cached.

Reply:
will adjust that

Reply:
Addressed with f496c2c